src/patterns/ec2-app.test.ts

```diff
@@ -68,6 +68,19 @@ describe("the GuEC2App pattern", function () {
     expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
   });
 
+  it("can produce an EC2 app with an internal load balancer", function () {
+    const stack = simpleGuStackForTesting();
+    new GuEc2App(stack, {
+      applicationPort: GuApplicationPorts.Node,
+      app: "test-gu-ec2-app",
+      access: { scope: AccessScope.INTERNAL, cidrRanges: [Peer.ipv4("10.0.0.0/8")] },
+      monitoringConfiguration: { noMonitoring: true },
+      userData: "#!/bin/dev foobarbaz",
+      certificateProps: getCertificateProps(),
+    });
+    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
```

Comment:
The snapshot for this test is pretty big (~900 lines), which makes it difficult to find what we're asserting. Could we use [`toHaveResourceOfTypeAndLogicalId`](https://github.com/guardian/cdk/blob/a5e1337b2b7f1d9ec1b5e1eb4e5b0adf0db1f393/src/utils/test/jest.ts#L42) to assert against the load balancer in isolation of the other resources that get created in the `GuEc2App`?